ada_feeding/ada_feeding/trees/acquire_food_tree.py

```diff
@@ -133,9 +173,171 @@ def create_tree(
+                            },
+                            outputs={
+                                "constraints": BlackboardKey("goal_constraints"),
+                            },
+                        ),
+                        MoveIt2Plan(
+                            name="MoveIntoPlan",
+                            ns=name,
+                            inputs={
+                                "goal_constraints": BlackboardKey("goal_constraints"),
+                                "max_velocity_scale": 0.8,
+                                "max_acceleration_scale": 0.8,
+                                "cartesian": True,
+                                "cartesian_max_step": 0.001,
+                                "cartesian_fraction_threshold": 0.95,
+                            },
+                            outputs={"trajectory": BlackboardKey("trajectory")},
+                        ),
+                        MoveIt2Execute(
+                            name="MoveInto",
+                            ns=name,
+                            inputs={"trajectory": BlackboardKey("trajectory")},
+                            outputs={},
+                        ),
+                        ### Extraction
+                        ComputeExtractConstraints(
+                            name="ComputeExtractConstraints",
+                            ns=name,
+                            inputs={
+                                "action": BlackboardKey("action"),
+                                # Default approach_frame_id = "approach"
+                            },
+                            outputs={
+                                "extract_position": BlackboardKey("extract_position"),
+                                "extract_orientation": BlackboardKey(
+                                    "extract_orientation"
+                                ),
+                                "ft_thresh": BlackboardKey("ft_thresh"),
+                                "ee_frame_id": BlackboardKey("ee_frame_id"),
+                            },
+                        ),
+                        MoveIt2PositionConstraint(
+                            name="ExtractPosition",
+                            ns=name,
+                            inputs={
+                                "position": BlackboardKey("extract_position"),
+                                "frame_id": "approach",
+                            },
+                            outputs={
+                                "constraints": BlackboardKey("goal_constraints"),
+                            },
+                        ),
+                        MoveIt2OrientationConstraint(
+                            name="ExtractOrientation",
+                            ns=name,
+                            inputs={
+                                "quat_xyzw": BlackboardKey("extract_orientation"),
+                                "frame_id": BlackboardKey("ee_frame_id"),
```

Comment:
I like how easy it is to specify that orientation and position are in different frames :)